src/test/kfs_inference_service_test.cpp

```diff
@@ -0,0 +1,431 @@
+                                               {1, 3, 62, 62},
+                                           }}}),
+            tensor_desc_map_t({{"Output_I32_1_2000", {
+                                                         ovms::Precision::I32,
+                                                         {1, 2000},
+                                                     }},
+                {"Output_FP32_2_20_3", {
+                                           ovms::Precision::FP32,
+                                           {2, 20, 3},
+                                       }}}));
+    }
+
+protected:
+    std::string modelName = "resnet";
+    ovms::model_version_t modelVersion = 23;
+
+    std::shared_ptr<NiceMock<MockModelInstance>> instance;
+    ::inference::ModelMetadataResponse response;
+    std::unique_ptr<ov::Core> ieCore;
+
+    void SetUp() override {
+        ieCore = std::make_unique<ov::Core>();
+    }
+
+    void TearDown() override {
+        ieCore.reset();
+    }
+};
+
+TEST_F(ModelMetadataResponseBuild, BasicResponseMetadata) {
+    prepare();
+
+    ASSERT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(instance, &response), ovms::StatusCode::OK);
+
+    EXPECT_EQ(response.name(), "resnet");
+
+    EXPECT_EQ(response.versions_size(), 1);
+    EXPECT_EQ(response.versions().at(0), "23");
+
+    EXPECT_EQ(response.platform(), "OpenVINO");
+}
+
+TEST_F(ModelMetadataResponseBuild, ModelVersionNotLoadedAnymore) {
+    prepare();
+    instance->retireModel();
+    EXPECT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(instance, &response), ovms::StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE);
+}
+
+TEST_F(ModelMetadataResponseBuild, ModelVersionNotLoadedYet) {
+    prepare();
+    instance->loadModel(DUMMY_MODEL_CONFIG);
+    EXPECT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(instance, &response), ovms::StatusCode::MODEL_VERSION_NOT_LOADED_YET);
+}
+
+TEST_F(ModelMetadataResponseBuild, SingleInputSingleOutputValidResponse) {
+    tensor_desc_map_t inputs = tensor_desc_map_t({{"SingleInput", {ovms::Precision::FP32, {1, 3, 224, 224}}}});
+    tensor_desc_map_t outputs = tensor_desc_map_t({{"SingleOutput", {ovms::Precision::I32, {1, 2000}}}});
+    prepare(inputs, outputs);
+
+    ASSERT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(instance, &response), ovms::StatusCode::OK);
+
+    EXPECT_EQ(response.inputs_size(), 1);
+    auto input = response.inputs().at(0);
+    EXPECT_EQ(input.name(), "SingleInput");
+    EXPECT_EQ(input.datatype(), "FP32");
+    EXPECT_EQ(input.shape_size(), 4);
+    EXPECT_TRUE(isShapeTheSame(input.shape(), {1, 3, 224, 224}));
+
+    EXPECT_EQ(response.outputs_size(), 1);
+    auto output = response.outputs().at(0);
+    EXPECT_EQ(output.name(), "SingleOutput");
+    EXPECT_EQ(output.datatype(), "I32");
+    EXPECT_EQ(output.shape_size(), 2);
+    EXPECT_TRUE(isShapeTheSame(output.shape(), {1, 2000}));
+}
+
+TEST_F(ModelMetadataResponseBuild, DoubleInputSingleOutputValidResponse) {
```

Comment:
I think we can skip those two tests without impact for test coverage:
SingleInputDoubleOutputValidResponse
DoubleInputSingleOutputValidResponse

Since this is covered somewhat enough in:
SingleInputSingleOutputValidResponse
DoubleInputDoubleOutputValidResponse